core/domain/wipeout_service.py

```diff
@@ -155,32 +158,67 @@ def pre_delete_user(user_id):
         subscribed_exploration_summaries = (
-            exp_fetchers.get_exploration_summaries_subscribed_to(
-                user_id)
-        )
+            exp_fetchers.get_exploration_summaries_subscribed_to(user_id))
+
         explorations_to_be_deleted_ids = [
             exp_summary.id for exp_summary in subscribed_exploration_summaries
             if exp_summary.is_private() and
-            exp_summary.is_solely_owned_by_user(user_id)]
+            exp_summary.is_solely_owned_by_user(user_id)
+        ]
         exp_services.delete_explorations(
             user_id, explorations_to_be_deleted_ids)
 
+        explorations_to_release_ownership_ids = [
+            exp_summary.id for exp_summary in subscribed_exploration_summaries
+            if not exp_summary.is_private() and
+            exp_summary.is_solely_owned_by_user(user_id)
+        ]
+        for exp_id in explorations_to_release_ownership_ids:
+            rights_manager.release_ownership_of_exploration(
+                user_services.get_system_user(), exp_id)
+
+        explorations_to_remove_user_from_ids = [
+            exp_summary.id for exp_summary in subscribed_exploration_summaries
+            if not exp_summary.is_solely_owned_by_user(user_id)
+        ]
+        for exp_id in explorations_to_remove_user_from_ids:
+            rights_manager.deassign_role_for_exploration(
+                user_services.get_system_user(), exp_id, user_id)
+
         subscribed_collection_summaries = (
-            collection_services.get_collection_summaries_subscribed_to(
-                user_id)
-        )
+            collection_services.get_collection_summaries_subscribed_to(user_id))
         collections_to_be_deleted_ids = [
             col_summary.id for col_summary in subscribed_collection_summaries
             if col_summary.is_private() and
-            col_summary.is_solely_owned_by_user(user_id)]
+            col_summary.is_solely_owned_by_user(user_id)
+        ]
         collection_services.delete_collections(
             user_id, collections_to_be_deleted_ids)
 
+        collections_to_release_ownership_ids = [
+            col_summary.id for col_summary in subscribed_collection_summaries
+            if not col_summary.is_private() and
+            col_summary.is_solely_owned_by_user(user_id)
+        ]
+        for col_id in collections_to_release_ownership_ids:
+            rights_manager.release_ownership_of_collection(
+                user_services.get_system_user(), col_id)
+
+        collections_to_remove_user_from_ids = [
+            col_summary.id for col_summary in subscribed_collection_summaries
+            if not col_summary.is_solely_owned_by_user(user_id)
+        ]
+        for col_id in collections_to_remove_user_from_ids:
+            rights_manager.deassign_role_for_collection(
+                user_services.get_system_user(), col_id, user_id)
+
+        topic_services.deassign_user_from_all_topics(
```

Comment:
Would also need to remove them from the general topic manager role, etc. -- right?